File: ELearningFrontEnd/src/app/dtos/chapter/chapter.req.ts
import {Lecture} from "../../model/lecture";
import {LectureReq} from "../lecture/lecture.req";

export class ChapterReq{

    name: string;
    courseId: number;
    lectures: LectureReq[] | null;

    constructor(data: any) {
        this.name = data.name;
        this.courseId = data.courseId;
        this.lectures = data.lectures
    }
}

File: ELearningBackEnd/src/main/java/com/example/elearningbackend/chapter/ChapterReq.java
package com.example.elearningbackend.chapter;

import com.example.elearningbackend.lecture.LectureReq;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class ChapterReq {

    private String name;

    private long courseId;

    private List<LectureReq> lectures = new ArrayList<>();

}
